rest/functionsapitest/user_functions_test.go

```diff
@@ -0,0 +1,1463 @@
+	})
+
+	t.Run("Delete a non-existing function and check HTTP Status Code", func(t *testing.T) {
+		response := rt.SendAdminRequest("DELETE", "/db/_config/functions/foo", "")
+		assert.Equal(t, 404, response.Result().StatusCode)
+	})
+}
+
+/// ILLEGAL SYNTAX TESTS
+
+func TestIllegalSyntax(t *testing.T) {
+	rt := rest.NewRestTesterForUserQueries(t, rest.DbConfig{})
+	if rt == nil {
+		return
+	}
+	defer rt.Close()
+
+	// positive cases
+	t.Run("n1ql queries lower and upper case", func(t *testing.T) {
+		var kN1QLFunctionsConfig = functions.FunctionsConfig{
+			Definitions: functions.FunctionsDefs{
+				"lowercase": &functions.FunctionConfig{
+					Type:  "query",
+					Code:  `seleCt $args.city AS city`,
+					Args:  []string{"city"},
+					Allow: allowAll,
+				},
+				"parens": &functions.FunctionConfig{
+					Type:  "query",
+					Code:  `  (select $args.city AS city)`,
+					Args:  []string{"city"},
+					Allow: allowAll,
+				},
+			},
+		}
+
+		request, err := json.Marshal(kN1QLFunctionsConfig)
+		assert.NoError(t, err)
+
+		response := rt.SendAdminRequest("PUT", "/db/_config/functions", string(request))
+		assert.Equal(t, 200, response.Result().StatusCode)
+	})
+
+	// negative cases
+	t.Run("bad js syntax", func(t *testing.T) {
+		response := rt.SendAdminRequest("PUT", "/db/_config/functions", `{
+			"definitions": {
+				"syntax_error": {
+					"type": "javascript",
+					"code": "returm )42(",
+					"allow": {"channels": ["*"]}}
+			}
+		}`)
+		assert.Equal(t, 400, response.Result().StatusCode)
+		assert.Contains(t, string(response.BodyBytes()), "Error compiling function")
+	})
+
+	t.Run("drop n1ql query", func(t *testing.T) {
+		// Can only register SELECT queries
+		response := rt.SendAdminRequest("PUT", "/db/_config/functions", `{
+			"definitions": {
+				"evil_n1ql_mutation": {
+					"type": "query",
+					"code": "DROP COLLECTION Students",
+					"allow": {"channels": ["*"]}}
+			}
+		}`)
+		assert.Equal(t, 400, response.Result().StatusCode)
+		assert.Contains(t, string(response.BodyBytes()), "only SELECT queries are allowed")
+	})
+
+	t.Run("select n1ql query with bad syntax", func(t *testing.T) {
+		// A bad SELECT query can be registered to config
+		// But, executing it will result in an error
+		response := rt.SendAdminRequest("PUT", "/db/_config/functions", `{
+			"definitions": {
+				"bad_n1ql_syntax": {
+					"type": "query",
+					"code": "SELECT )22( AS Students :",
+					"allow": {"channels": ["*"]}}
+			}
+		}`)
+		assert.Equal(t, 200, response.Result().StatusCode)
+
+		response = rt.SendAdminRequest("GET", "/db/_function/bad_n1ql_syntax", "")
+		assert.Equal(t, 500, response.Result().StatusCode)
+		assert.Contains(t, string(response.BodyBytes()), "Internal Server Error")
+		assert.Contains(t, string(response.BodyBytes()), "syntax error")
+	})
+}
+
+//////// FUNCTIONS EXECUTION API TESTS
+
+/// AUTH TESTS
+
+func createUserAlice(t *testing.T, rt *rest.RestTester) (string, string) {
```

Comment:
All of this from here to line 1200 or so is just copied-and-pasted, with adaptations, from the functions package. Testing it again here is redundant.